src/storage/src/source/mysql/snapshot.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::rc::Rc;
+use std::str::FromStr;
+
+use differential_dataflow::{AsCollection, Collection};
+use futures::TryStreamExt;
+use mysql_async::prelude::Queryable;
+use mysql_async::{IsolationLevel, Row as MySqlRow, TxOpts};
+use mz_sql_parser::ast::display::AstDisplay;
+use timely::dataflow::channels::pact::Pipeline;
+use timely::dataflow::operators::{Broadcast, CapabilitySet, Concat, ConnectLoop, Feedback, Map};
+use timely::dataflow::{Scope, Stream};
+use timely::progress::{Antichain, Timestamp};
+use tracing::trace;
+
+use mz_mysql_util::{
+    ensure_full_row_binlog_format, ensure_gtid_consistency, ensure_replication_commit_order,
+    query_sys_var, GtidSet, SchemaRequest,
+};
+use mz_mysql_util::{schema_info, MySqlTableDesc};
+use mz_repr::{Diff, GlobalId, Row};
+use mz_sql_parser::ast::UnresolvedItemName;
+use mz_storage_types::sources::MySqlSourceConnection;
+use mz_timely_util::builder_async::{
+    Event as AsyncEvent, OperatorBuilder as AsyncOperatorBuilder, PressOnDropButton,
+};
+
+use crate::source::{RawSourceCreationConfig, SourceReaderError};
+
+use super::{pack_mysql_row, ReplicationError, RewindRequest, TransactionId, TransientError};
+
+/// Renders the snapshot dataflow. See the module documentation for more information.
+pub(crate) fn render<G: Scope<Timestamp = TransactionId>>(
+    mut scope: G,
+    config: RawSourceCreationConfig,
+    connection: MySqlSourceConnection,
+    subsource_resume_uppers: BTreeMap<GlobalId, Antichain<TransactionId>>,
+    table_info: BTreeMap<UnresolvedItemName, (usize, MySqlTableDesc)>,
+) -> (
+    Collection<G, (usize, Result<Row, SourceReaderError>), Diff>,
+    Stream<G, RewindRequest>,
+    Stream<G, ReplicationError>,
+    PressOnDropButton,
+) {
+    let mut builder =
+        AsyncOperatorBuilder::new(format!("MySqlSnapshot({})", config.id), scope.clone());
```

Comment:
Just for consistency with the other name
```suggestion
        AsyncOperatorBuilder::new(format!("MySqlSnapshotReader({})", config.id), scope.clone());
```